### PR TITLE
HFA: Fix some typos in HFA -- Erdas Imagine .img docs

### PR DESCRIPTION
**Pull request by echoix**
## What does this PR do?

I had to read the page https://gdal.org/drivers/raster/hfa.html and noticed some typos in the page.
I fixed the ones that I understood.

## Tasklist

 - [ ] Please help reword the paragraph I linked to below, it is unclear. I included in this PR at least is->are, but it still doesn't make sense
       https://github.com/OSGeo/gdal/blob/ced6a0ded84a892aefcdbf449689026e3645494e/doc/source/drivers/raster/hfa.rst?plain=1#L22-L26
 - [ ] Make sure code is correctly formatted (cf [pre-commit configuration](https://gdal.org/development/dev_practices.html#commit-hooks))
 - [ ] Add test case(s)
 - [ ] Add documentation
 - [ ] Updated Python API documentation (swig/include/python/docs/)
 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

## Environment

Provide environment details, if relevant:

* OS:
* Compiler:

### CODE REVIEW

**Comment by rouault** [comment]
> * Please help reword the paragraph I linked to below, it is unclear

yeah, quite cryptic. This dates back to the origins: https://github.com/OSGeo/gdal/commit/72f5bb8947d0f22beab282202f25dada3ced70ad

Perhaps the following would be slightly clearer?
```Metadata reading and writing is supported at the dataset and band level. But this is GDAL specific metadata, not metadata in an Imagine recognized form. The metadata is stored in a table called GDAL_MetaData which as many columns as metadata items. The title of the column is the {key} of the {key}={value} metadata item pair and the value of row 1 is {value}.```

**Comment by echoix** [comment]
Also, since I wasn't familiar at all with this driver (got linked to in the grass-user mailing list), I had to look up about `ArcGIS PE String`, but it seems it might have changed since. Is it the same thing as `ESRI Projection Engine (PE) String` from https://support.esri.com/en-us/knowledge-base/why-does-the-spatial-reference-change-when-exporting-da-000011610 (PE mention) and https://github.com/Esri/projection-engine-db-doc (ESRI projection engine reference)

**Comment by echoix** [comment]
> > * Please help reword the paragraph I linked to below, it is unclear
> 
> yeah, quite cryptic. This dates back to the origins: [72f5bb8](https://github.com/OSGeo/gdal/commit/72f5bb8947d0f22beab282202f25dada3ced70ad)
> 
> Perhaps the following would be slightly clearer? `Metadata reading and writing is supported at the dataset and band level. But this is GDAL specific metadata, not metadata in an Imagine recognized form. The metadata is stored in a table called GDAL_MetaData which as many columns as metadata items. The title of the column is the {key} of the {key}={value} metadata item pair and the value of row 1 is {value}.`

I assume the curly braces are reStructuredText syntax, right? It's been ages since I've written in this markup.

**Comment by rouault** [comment]
> I assume the curly braces are reStructuredText syntax, right?

no, I just meant them to indicate substituted values, but that might perhaps not be necessary for the understanding.

**Comment by rouault** [comment]
> I had to look up about `ArcGIS PE String`, but it seems it might have changed since. Is it the same thing as `ESRI Projection Engine (PE) String`

yes

File: doc/source/drivers/raster/hfa.rst [hunk 1]
```diff
@@ -16,14 +16,15 @@ and c128.
...
-Metadata reading and writing is supported at the dataset level, and for
-bands, but this is GDAL specific metadata - not metadata in an Imagine
-recognized form. The metadata is stored in a table called GDAL_MetaData
-with each column being a metadata item. The title is the key and the row
-1 value is the value.
+Metadata reading and writing is supported at the dataset and band level.
+But this is GDAL specific metadata, not metadata in an Imagine recognized form.
+The metadata is stored in a table called GDAL_MetaData which as many 
+columns as metadata items. 
+The title of the column is the {key} of the {key}={value} metadata item pair
```
Comment (echoix) [inline doc/source/drivers/raster/hfa.rst:26]: @rouault oups, I left them there. Is there another syntax you'd like?

File: doc/source/drivers/raster/hfa.rst [hunk 2]
```diff
@@ -16,14 +16,15 @@ and c128.
...
+and the value of row 1 is {value}.
```
Comment (rouault) [inline doc/source/drivers/raster/hfa.rst:27]: Perhaps?
```suggestion
The title of the column is the ``key`` of the ``key``=``value`` metadata item pair
and the value of row 1 is ``value``.
```

Comment (echoix) [reply doc/source/drivers/raster/hfa.rst]: Yep! Thanks!

**Pull request merged**